custom_metrics/javascript.js

```diff
@@ -0,0 +1,96 @@
+                ajax_h2: ajaxs.filter(r => ['h2', 'http/2', 'http2'].includes(r.nextHopProtocol)).length,
+                resources_h2: resources.filter(r => ['h2', 'http/2', 'http2'].includes(r.nextHopProtocol)).length,
+                ajax_h3: ajaxs.filter(r => ["h3", "http/3", "h3-29"].includes(r.nextHopProtocol)).length,
+                resources_h3: resources.filter(r => ["h3", "http/3", "h3-29"].includes(r.nextHopProtocol)).length,
+            }
+            return request_data;
+        } catch (e) {
+            return null;
+        }
+    })(),
+
+    web_component_specs: (() => {
+
+        var getNodeName = (el) => el.nodeName.toLowerCase()
+        var elements_with_hyphen = Array.from(document.getElementsByTagName("*")).filter(e => e.nodeName.includes('-'));
+        var unique_elements_with_hyphen = [...new Set(elements_with_hyphen.map(e => e.nodeName))].map(element => { return elements_with_hyphen.find(e => e.nodeName === element) });
+        var web_components = unique_elements_with_hyphen.filter(e => customElements.get(e.nodeName.toLowerCase()));
+
+        // Checking which web comp spec is used by found web_components (template, shadow_dom, custom_elements)
+        var shadow_roots = web_components.filter(e => e.shadowRoot)
+        var template = web_components.filter(e => Array.from(e.children).some(r => r.nodeName.toLowerCase() === 'template'))
+        var web_component_specs = {
+            'custom_elements': web_components.map(getNodeName),
+            'shadow_roots': shadow_roots.map(getNodeName),
+            'template': template.map(getNodeName)
+        }
+        return web_component_specs
+    }),
+
+    script_tags: (() => {
+        let script_tags = Array.from(document.querySelectorAll('script'));
+
+        let script_data = {
+            total: script_tags.length,
+            async: script_tags.filter(tag => tag.async).length,
+            defer: script_tags.filter(tag => tag.defer).length,
+            crossorigin: script_tags.filter(tag => tag.crossorigin).length,
+            integrity: script_tags.filter(tag => tag.integrity).length,
+            nomodule: script_tags.filter(tag => tag.nomodule).length,
+            nonce: script_tags.filter(tag => tag.nonce).length,
+            referrerpolicy: script_tags.filter(tag => tag.referrerpolicy).length,
+            src: script_tags.filter(tag => tag.src).length,
+            inline: script_tags.filter(tag => !tag.src).length,
+            type_module: script_tags.filter(tag => tag.type == 'module').length,
+            async_and_defer: script_tags.filter(tag => tag.async && tag.defer).length,
+            defer_without_src: script_tags.filter(tag => tag.defer && !tag.src).length,
+            //deprecated attribute adoption
+            charset: script_tags.filter(tag => tag.charset).length,
+            language: script_tags.filter(tag => tag.attributes['language']).length,
```

Comment:
A bit more semantic (could be done for all of these attribute metrics, but seems like `language` is a special case)
```suggestion
            language: script_tags.filter(tag => tag.hasAttribute('language')).length,
```

Reply:
there we go!